packages/interface-connection-encrypter/src/index.ts

```diff
@@ -5,22 +5,23 @@ import type { Duplex } from 'it-stream-types'
  * A libp2p connection encrypter module must be compliant to this interface
  * to ensure all exchanged data between two peers is encrypted.
  */
-export interface ConnectionEncrypter {
+export interface ConnectionEncrypter<Extension = any> {
   protocol: string
 
   /**
    * Encrypt outgoing data to the remote party.
    */
-  secureOutbound: (localPeer: PeerId, connection: Duplex<Uint8Array>, remotePeer: PeerId) => Promise<SecuredConnection>
+  secureOutbound: (localPeer: PeerId, connection: Duplex<Uint8Array>, remotePeer: PeerId) => Promise<SecuredConnection<Extension>>
 
   /**
    * Decrypt incoming data.
    */
-  secureInbound: (localPeer: PeerId, connection: Duplex<Uint8Array>, remotePeer?: PeerId) => Promise<SecuredConnection>
+  secureInbound: (localPeer: PeerId, connection: Duplex<Uint8Array>, remotePeer?: PeerId) => Promise<SecuredConnection<Extension>>
 }
 
-export interface SecuredConnection {
+export interface SecuredConnection<E> {
```

Comment:
I used a generic here to avoid pulling in the protobuf definition.

Reply:
I don't think you need to pull protobuf definition, you can define an interface in this file:

interface NoiseExtensions {
  webtransportCerthashes: Uint8Array[]
}

Reply:
This means you have to keep this up to date with the generated code from the protobuf. Doesn't seem ideal, no?

Reply:
This is an interface, so it should ensure all implementations get a type error in case they are incompatible with the version of interface.

Reply:
Ah, on second thought, you are right, it probably makes no sense to keep this interface noise specific.
We should make `remoteExtensions` optional and change SecuredConnection into `export interface SecuredConnection<E = unknown>`

Reply:
Updated. But does it make sense to set the default generic param for SecuredConnection? I don't think we need to